### PR TITLE
SrcUriFilenameDotPrefix: new check for SRC_URI filenames with dot prefix

### PR DESCRIPTION
**Pull request by arthurzam**
Resolves: https://github.com/pkgcore/pkgcheck/issues/585

### CODE REVIEW

**Review by thesamesam — APPROVED** [review]
Nice and straightforward. Thanks!

**Comment by mgorny** [comment]
I wonder if we shouldn't simply put it in `bad_filenames`.

**Comment by arthurzam** [comment]
> I wonder if we shouldn't simply put it in `bad_filenames`.

@mgorny I'm unsure about it. While simpler, the BadFilename is quite ambiguous, and I'm unsure anybody will understand what it means in practice. Here we have a more specialized case, where we warn the develop - "you might have done big mistake, reverify".

I think it will be better to have more info than less here.

File: src/pkgcheck/checks/metadata.py [hunk 1]
```diff
@@ -1315,6 +1315,23 @@ class InvalidSrcUri(results.MetadataError, results.VersionResult):
     attr = "fetchables"
 
 
+class SrcUriFilenameDotPrefix(results.VersionResult, results.Warning):
```
Comment (mgorny) [inline src/pkgcheck/checks/metadata.py:1318]:
```suggestion
class SrcUriFilenameDotPrefix(results.VersionResult, results.Error):
```

**Review by mgorny — COMMENTED** [review]
Yeah, makes sense — but let's make this one fatal though.

File: src/pkgcheck/checks/metadata.py (hunk 1 above)
Comment (arthurzam) [reply src/pkgcheck/checks/metadata.py]: Good idea, done

**Review by mgorny — APPROVED** [review]
LGTM

**Pull request merged**